api-matlab/wrapper_physiobatch.m

```diff
@@ -0,0 +1,111 @@
+        throw(ME);
+    end
+    
+    % read scan parameters
+    input_params = inputdlg({'Number of scans:','Number of slices:','Number of dummies:', 'TR:', 'Onset slice:'}, 'Parameters for physio');
+    
+    % scan parameters
+    matlabbatch{1}.spm.tools.physio.log_files.sampling_interval = [];
+    matlabbatch{1}.spm.tools.physio.log_files.relative_start_acquisition = 0;
+    matlabbatch{1}.spm.tools.physio.log_files.align_scan = 'last'; % sometimes first, sometimes last
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nslices = str2double(input_params{2});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.NslicesPerBeat = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.TR = str2double(input_params{4});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Ndummies = str2double(input_params{3});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nscans = str2double(input_params{1});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.onset_slice = str2double(input_params{5});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.time_slice_to_slice = [];
+
+    % template
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nprep = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sync.nominal = struct([]);
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.modality = 'ECG';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.min = 0.4;
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.file = 'initial_cpulse_kRpeakfile.mat';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.posthoc_cpulse_select.off = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.output_multiple_regressors = 'multiple_regressors.txt';
+    matlabbatch{1}.spm.tools.physio.model.output_physio = 'physio.mat';
+    matlabbatch{1}.spm.tools.physio.model.orthogonalise = 'none';
+    matlabbatch{1}.spm.tools.physio.model.censor_unreliable_recording_intervals = false;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.c = 3;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.r = 4;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.cr = 1;
+    matlabbatch{1}.spm.tools.physio.model.rvt.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.hrv.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.noise_rois.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.movement.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.other.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.verbose.level = 2;
+    matlabbatch{1}.spm.tools.physio.verbose.fig_output_file = '';
+    matlabbatch{1}.spm.tools.physio.verbose.use_tabs = false;
+    
+    % save batch
+     save([out_dir filesep 'physio_batch'], 'matlabbatch');
+     disp("Saved matlabbatch in:");
+     disp(out_dir);
+        
+    % spm_jobman('run',matlabbatch);
```

Comment:
If this line is uncommented, am I correct in understanding that the resulting batch will generate the regressors and nothing else? Or does this also trigger further processes with Physio?

Reply:
If this line is uncommented, it starts the whole physIO processing and generate regressors as well as plots showing the regressors and signals over time. I remembered from our Skype conversation that we decided just to save the batch, but we could also include this line such that physio processing is directly executed.  That's a design choice on how much you want to have rtQC do automatically.

Reply:
Ok, so as discussed today we will let this run. If Physio is not installed it will error out. If installed, it will:

- run the batch
- generate the regressors
- fetch useful timeseries and plots from Physio and plot in rtQC
- add the same useful information to the HTML report